src/components/networked-hand-controls.js

```diff
@@ -153,38 +113,284 @@ AFRAME.registerComponent('networked-hand-controls', {
+    // this block is for receiving+handling networked hand gestures and button events for controller model updates
+    // and for initial injection of remote controller model
+    if (!this.local) {
+      if (this.data.handModelStyle !== this.str.controller && this.data.gesture !== oldData.gesture) {
+        this.handleGesture(this.data.gesture, oldData.gesture);
+      }
+      else if (this.data.handModelStyle === this.str.controller) {
+        if (!this.injectedController && this.data.controllerComponent && this.data.webxrControllerProfiles[0]) {
+          this.injectRemoteControllerModel();
+        }
+        if (oldData.controllerEvent && oldData.controllerEvent !== this.data.controllerEvent) {
+          const controllerEvent = JSON.parse(this.data.controllerEvent);
+          this.el.components[this.data.controllerComponent][controllerEvent.type === 'thumbstickmoved' ? 'onThumbstickMoved' : 'onButtonChanged'](controllerEvent);
+        }
+        if (oldData.controllerModelUpdate && (
+          oldData.controllerModelUpdate[0] !== this.data.controllerModelUpdate[0] ||
+          oldData.controllerModelUpdate[1] !== this.data.controllerModelUpdate[1] 
+        )) {
+          this.el.components[this.data.controllerComponent].updateModel(...this.data.controllerModelUpdate)
+        }
+      }
     }
   },
 
   remove() {
-    this.el.removeObject3D('mesh');
+    this.el.removeObject3D(this.str.mesh);
+    this.removeEventListeners();
   },
-  
-  // cache system, protection from A-Frame optimization behavior while reducing garbage collection overhead  
-  Z: 0,
-  Y: {},
-  
+
+  addHandModel() {
+    this.loader = new THREE.GLTFLoader();
+    this.loader.setCrossOrigin('anonymous');
+    const handmodelUrl = this.MODEL_BASE + this.MODEL_NAMES[this.data.handModelStyle + this.data.hand.charAt(0).toUpperCase() + this.data.hand.slice(1)];
+    
+    this.loader.load(this.data.customHandModelURL || handmodelUrl, gltf => {
+      const newMesh = gltf.scene.children[0];
+      const handModelOrientation = this.data.hand === 'left' ? Math.PI / 2 : -Math.PI / 2;
+      newMesh.mixer = new THREE.AnimationMixer(newMesh);
+
+      this.clips = gltf.animations;
+      this.clips.forEach((clip) => {
+        this.clipNameToClip[clip.name] = clip;
+      })      
+
+      this.el.setObject3D(this.str.mesh, newMesh);
+
+      const handMaterial = newMesh.children[1].material;
+      handMaterial.color = new THREE.Color(this.data.handColor);
+      newMesh.position.set(0, 0, 0);
+      newMesh.rotation.set(0, 0, handModelOrientation);
+
+      this.updateHandMeshColor();
+    });
+  },
+
+  updateHandMeshColor() {
+    const color = new THREE.Color(this.data.color);
+
+    // a different way to set color that we aren't using:
+    // this.getMesh().children[1].material.emissive = color;
+    // this.getMesh().children[1].material.color = color;
+    // this.getMesh().children[1].material.metalness = 1;
```

Comment:
Can you remove those comments? Also you don't need to create a THREE.Color instance here, `color.set(this.data.color)` should be fine with any string.